drivers/spi/spi_context.h

```diff
@@ -154,33 +154,34 @@ static inline void spi_context_cs_control(struct spi_context *ctx, bool on)
 	}
 }
 
-static inline void spi_context_buffers_setup(struct spi_context *ctx,
-					     const struct spi_buf *tx_bufs,
-					     size_t tx_count,
-					     struct spi_buf *rx_bufs,
-					     size_t rx_count,
-					     u8_t dfs)
+static inline
+void spi_context_buffers_setup(struct spi_context *ctx,
+			       const struct spi_buf_array *tx_bufs,
+			       const struct spi_buf_array *rx_bufs,
+			       u8_t dfs)
 {
-	SYS_LOG_DBG("tx_bufs %p (%zu) - rx_bufs %p (%zu) - %u",
-		    tx_bufs, tx_count, rx_bufs, rx_count, dfs);
-
-	ctx->current_tx = tx_bufs;
-	ctx->tx_count = tx_count;
-	ctx->current_rx = rx_bufs;
-	ctx->rx_count = rx_count;
+	SYS_LOG_DBG("tx_bufs %p - rx_bufs %p - %u", tx_bufs, rx_bufs, dfs);
 
 	if (tx_bufs) {
-		ctx->tx_buf = tx_bufs->buf;
-		ctx->tx_len = tx_bufs->len / dfs;
+		ctx->current_tx = (struct spi_buf *) tx_bufs->bufs;
```

Comment:
You really should not do things like that. `bufs` is declared as pointing to something that will not be modified. This also protects against "accidental" modifications of the pointed content (the compiler will object), which is good - an attempt to write to a structure for example placed in flash will not end well. Why to remove such protection?

Reply:
That's what I tried to tell in the meeting last evening: if I don't do that (which is crap, I agree), compilation gives a warning there.

I am incrementing this pointer in spi_context.h (ctx->current_tx that is).

Reply:
Yes, you are incrementing the pointer, so only its value is changed (i.e. it is moved to the next buffer), but the pointed content is not modified. That's why both `current_tx` and `current_rx` should be pointers to constant structures (but not constant pointers).

Reply:
confusing, I was sure I got warnings when having current_tx/rx as const, but no. 

Ok, solved then